src/Tensors/blocksparse/blocksparse.jl

```diff
@@ -72,8 +72,17 @@ end
 # TODO: this could be a generic TensorStorage function
 Base.complex(D::BlockSparse{T}) where {T} = BlockSparse{complex(T)}(complex(data(D)),
                                                                     blockoffsets(D))
-Base.conj(D::BlockSparse{<: Real}) = D
-Base.conj(D::BlockSparse) = BlockSparse(conj(data(D)), copy(blockoffsets(D)))
+
+function Base.conj(D::BlockSparse{<: Real}; always_copy = false) 
+  if always_copy
+    return conj!(copy(D))
+  end
+  return D
+end
+
+Base.conj(D::BlockSparse; kwargs...) = BlockSparse(conj(data(D)), copy(blockoffsets(D)))
```

Comment:
Looks good, thanks. One thing this made me wonder about is if this should copy the block-offset list. Copying the block-offset list, in the limit of small blocks, could be as expensive as copying the data. Perhaps it should be `opt-in` with the `always_copy` keyword, and removed from this version of `conj`.

Reply:
Agreed; the latest commit below should fix

Reply:
I ran the tests manually and they are passing